packages/create-cloudflare/src/templates.ts

```diff
@@ -412,3 +436,25 @@ export const getTemplatePath = (ctx: C3Context) => {
 
 	return resolve(__dirname, "..", "templates", ctx.template.id);
 };
+
+export const isVariantInfo = (
+	copyFiles: CopyFiles
+): copyFiles is VariantInfo => {
+	return "path" in (copyFiles as VariantInfo);
+};
+
+export const getCopyFilesDestinationDir = async (
+	ctx: C3Context
+): Promise<undefined | string> => {
+	const { copyFiles } = ctx.template;
+
+	if (!copyFiles?.destinationDir) {
+		return undefined;
+	}
+
+	if (typeof copyFiles.destinationDir === "string") {
+		return copyFiles.destinationDir;
+	}
+
+	return copyFiles.destinationDir(ctx);
```

Comment:
Same question here, does this need to be async?

Reply:
removed 🙂👍